### PR TITLE
Keyboard Manager Component

### PR DESCRIPTION
**Pull request by wbreza**
Allows a single place to register keyboard events across the application.  Automatically registers and deregisters event handlers as keyboard bindings go int/out of scope.

- New `KeyboardManager` component (top level context component)
- New `KeyboardBinding` component (Allows keyboard bindings within pages / components)

### CODE REVIEW

File: src/react/components/common/keyboardBinding/keyboardBinding.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import React from "react";
+
+export interface IKeyboardBindingProps {
+    accelerator: string;
+    onKeyDown: (evt?: KeyboardEvent) => void;
+}
+
+export class KeyboardBinding extends React.Component<IKeyboardBindingProps> {
+    public static contextType = KeyboardContext;
+    public context!: IKeyboardContext;
```
Comment (elizabethhalper) [inline src/react/components/common/keyboardBinding/keyboardBinding.tsx]: Is this supposed to be a question mark or does the bang operator do something here? If so, what is it doing?

Comment (wbreza) [reply src/react/components/common/keyboardBinding/keyboardBinding.tsx]: https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci

File: src/react/components/toolbar/previousAsset.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { ToolbarItem } from "./toolbarItem";
```
Comment (pjlittle) [inline src/react/components/toolbar/previousAsset.tsx]: likely more trouble than what it's worth, but looking at previous/next in terms of DRY, an AssetNav component that takes direction?

Comment (wbreza) [reply src/react/components/toolbar/previousAsset.tsx]: Totally makes sense 👍

File: src/registerToolbar.ts [hunk 3]
```diff
@@ -70,6 +72,24 @@ export default function registerToolbar() {
         type: ToolbarItemType.Action,
     });
 
+    ToolbarItemFactory.register(PreviousAsset, {
+        name: "navigatePreviousAsset",
+        tooltip: "Previous Asset",
```
Comment (pjlittle) [inline src/registerToolbar.ts]: good for now, but we should pipe localization in here.

**Comment by pjlittle** [comment]
Good stuff, just left a couple minor remarks. Digging these components and centralized, data-driven registration.

File: src/registerToolbar.ts (hunk 3 above)
Comment (wbreza) [reply src/registerToolbar.ts]: 👍

**Pull request merged**